### ISSUE TITLE
Custom Toolbar Buttons

### ISSUE THREAD

**Issue by supervanya** [dromru/react-photoswipe-gallery#694]
**Is your feature request related to a problem? Please describe.**
I see the new version of photo swipe has ability to [add custom buttons](https://photoswipe.com/v5/docs/adding-custom-buttons/) to the toolbar, is this something that we can do in react photoswipe?

**Describe the solution you'd like**
Maybe something like this:

```js
import PhotoswipeUIDefault from 'photoswipe/dist/photoswipe-ui-default'
import { CustomGallery, Item, DefaultLayout } from 'react-photoswipe-gallery'

const MyGallery = () => {
  const layoutRef = useRef()

  return (
    <CustomGallery layoutRef={layoutRef} ui={PhotoswipeUIDefault}>
      {/*...*/}
    </CustomGallery>

    <CustomGallery layoutRef={layoutRef} ui={PhotoswipeUIDefault}>
      {/*...*/}
    </CustomGallery>

    <DefaultLayout
      shareButton={false}
      fullscreenButton={false}
      zoomButton={false}
      customButtons={[
          {
            name: 'edit-image',
            order: 2,
            isButton: true,
            icon: <Search />,
            html: 'Test',
            onClick: (event, el) => {
              console.log('clicked element ' + el)
            }
          }
      ]}
      ref={layoutRef}
    />
  )
}
```

**Describe alternatives you've considered**
Maybe even better solution would be to allow buttons to be react components and give them access to the PhotoSwipe API through render props / hook?

**Additional context**
I am deciding between using this library and React Image Lightbox and this feature is kind of a deal breaker at the moment.

**Comment by irudoy** [comment]
PhotoSwipe V5 is in the beta stage now and will be officially supported right after the stable release. We will try to expose most of the APIs in a React-ish way, of course.

For now, you can add custom buttons with your own modified Layout based on [this](https://github.com/dromru/react-photoswipe-gallery/blob/master/src/photoswipe-layout.tsx)

**Comment by jgujgu** [comment]
@irudoy I'm interested in using v5's custom button functionality. Has the beta yet moved to the point where this is easily useable?

**Comment by irudoy** [comment]
@jgujgu [PhotoSwipe](https://github.com/dimsemenov/photoswipe) v5 is still not released yet, we are also waiting

**Comment by irudoy** [comment]
Should be fixed in v2.1.0

https://github.com/dromru/react-photoswipe-gallery#custom-ui-elements

But `icon` and `html` should be strings, unfortunately. In a theory, we can do something as we did with custom content via portals, but it is a little bit tricky to do without much performance hit. We will implement it in the future, maybe.
